src/core/src/op/swish.cpp

```diff
@@ -50,81 +59,52 @@ void op::v4::Swish::validate_and_infer_types() {
     if (new_args.size() == 1) {
-        return make_shared<op::v4::Swish>(new_args.at(0));
+        return std::make_shared<Swish>(new_args.at(0));
     } else {
-        return make_shared<op::v4::Swish>(new_args.at(0), new_args.at(1));
+        return std::make_shared<Swish>(new_args.at(0), new_args.at(1));
     }
 }
 
-OPENVINO_SUPPRESS_DEPRECATED_START
-namespace swish {
-namespace {
-template <element::Type_t ET>
-inline bool evaluate(const HostTensorPtr& arg0,
-                     const HostTensorPtr& arg1,
-                     const HostTensorPtr& out,
-                     const size_t count) {
-    using T = typename element_type_traits<ET>::value_type;
-    if (arg1 != nullptr) {
-        ov::reference::swish<T>(arg0->get_data_ptr<ET>(), arg1->get_data_ptr<ET>(), out->get_data_ptr<ET>(), count);
-    } else {
-        ov::reference::swish<T>(arg0->get_data_ptr<ET>(), nullptr, out->get_data_ptr<ET>(), count);
-    }
-    return true;
-}
-
-bool evaluate_swish(const HostTensorVector& inputs, const HostTensorPtr& out) {
-    bool rc = true;
-    size_t count = shape_size(inputs[0]->get_shape());
-
-    const HostTensorPtr arg0 = inputs[0];
-    const HostTensorPtr arg1 = inputs.size() == 2 ? inputs[1] : nullptr;
-    out->set_unary(arg0);
-
-    switch (arg0->get_element_type()) {
-        NGRAPH_TYPE_CASE(evaluate_swish, f16, arg0, arg1, out, count);
-        NGRAPH_TYPE_CASE(evaluate_swish, f32, arg0, arg1, out, count);
-    default:
-        rc = false;
-        break;
-    }
-    return rc;
-}
-}  // namespace
-}  // namespace swish
-
-bool op::v4::Swish::evaluate(const HostTensorVector& outputs, const HostTensorVector& inputs) const {
+bool Swish::evaluate(TensorVector& outputs, const TensorVector& inputs) const {
     OV_OP_SCOPE(v4_Swish_evaluate);
-    OPENVINO_SUPPRESS_DEPRECATED_START
-    NGRAPH_CHECK(validate_host_tensor_vector(outputs, 1) &&
-                 (validate_host_tensor_vector(inputs, 2) || validate_host_tensor_vector(inputs, 1)));
-    OPENVINO_SUPPRESS_DEPRECATED_END
-    return swish::evaluate_swish(inputs, outputs[0]);
+    OPENVINO_ASSERT(outputs.size() == 1);
+    OPENVINO_ASSERT(swish::has_1_or_2_inputs(inputs.size()));
+
+    outputs[0].set_shape(inputs[0].get_shape());
+    const auto& arg1 = inputs.size() == 2 ? inputs[1] : Tensor();
+
+    using namespace ov::element;
+    return IfTypeOf<f16, f32>::apply<swish::Evaluate>(inputs[0].get_element_type(),
+                                                      inputs[0],
+                                                      arg1,
+                                                      outputs[0],
+                                                      shape_size(inputs[0].get_shape()));
 }
 
-bool op::v4::Swish::has_evaluate() const {
+bool Swish::has_evaluate() const {
     OV_OP_SCOPE(v4_Swish_has_evaluate);
     switch (get_input_element_type(0)) {
-    case ngraph::element::f16:
-    case ngraph::element::f32:
+    case element::f16:
```

Comment:
Should it be dropped in context of PR #19830?

Reply:
I've tried to drop it but then there are functional test failures in template and CPU plugin. And not sure if is enough to remove this precision from these tests or ir is required. This is reason why this drop is not introduced here and I think test issues will be resolved in #19830.